### DISCUSSION TITLE
Constrain recognition to a BNF-style grammar for command and control?

### DISCUSSION THREAD

**Discussion by ulatekh** [openai/whisper#2100] [Q&A]
Is it possible to constrain what Whisper recognizes to a BNF-style grammar, for command-and-control purposes?

If it's not presently implemented, can anyone speculate on where this could be integrated into the current code base?

Another way to look at it would be for Whisper to make available what it's recognized in the audio-sample so far (starting with nothing), and accepting a list of words it's allowed to recognize next, constraining its transcription to just those words.

**Comment by phineas-pta** [comment]
transcript post-processing seem to be the only easy way

the way u suggest would require train a model on top of whisper to do something like force words output

**Comment by ulatekh** [comment]
[whisper.cpp](https://github.com/ggerganov/whisper.cpp) has a grammar feature that works pretty well for me; I don't know if its technique can be backported to this version.
